Ensure resources (#2)
commercetools/commercetools-subscriptions: docs/HowToRun.md

# How to run

## Commercetools project requirements

Resources below are required for the commercetools-subscriptions to operate correctly.

### Types

> 1. [Checkout order type](../resources/checkout-order-type.json)

Reviewer: ```suggestion
1. [Checkout order](../resources/checkout-order-type.json)
```

Author: https://github.com/commercetools/commercetools-subscriptions/pull/2/commits/790cb77cc1be61e559b2584c49469a03dc61097d

1. [Checkout order line item type](../resources/checkout-order-line-item-type.json)
1. [Subscription order type](../resources/subscription-order-type.json)
1. [Subscription template order type](../resources/subscription-template-order-type.json)

### States
1. [Active state](../resources/active-state.json)

> 1. [Cancelled state](../resources/cancelled-state.json)

Reviewer: ```suggestion
1. [Cancelled](../resources/cancelled-state.json)
```

Reviewer: I think "state" is not needed for the link-label

Author: https://github.com/commercetools/commercetools-subscriptions/pull/2/commits/790cb77cc1be61e559b2584c49469a03dc61097d

1. [Error state](../resources/error-state.json)
1. [Paused state](../resources/paused-state.json)
1. [Reminder sent state](../resources/reminder-sent-state.json)
1. [Send reminder state](../resources/send-reminder-state.json)

> You can create these resources by running the command npm run setup-resources as below.

Reviewer: ```suggestion
You can create these resources by running the command 'npm run setup-resources' as below.
```

Author: https://github.com/commercetools/commercetools-subscriptions/pull/2/commits/da8a2919c14dcf5b9df7b3f649d528425ec4755e

```
export CTP_PROJECT_KEY=xxx
export CTP_CLIENT_ID=xxx
export CTP_CLIENT_SECRET=xxx
export EXISTING_ORDER_TYPE_KEY=xxx 
export EXISTING_ORDER_LINE_ITEM_TYPE_KEY=xxx 
npm run setup-resources
```

| Name | Content | Required |
| --- | --- | --- |
| CTP_PROJECT_KEY | commercetools project key | YES |
| CTP_CLIENT_ID | OAuth 2.0 client_id and can be used to obtain a token. | YES |
| CTP_CLIENT_SECRET | OAuth 2.0 client_secret and can be used to obtain a token. | YES |

> | EXISTING_ORDER_TYPE_KEY | Key of the existing order custom type. If provided, the script will merge `fieldDefinitions` from this type to the checkout order type. Use this option if you already have an order custom type and you want to use the attributes from this type with the checkout order type | NO |

Reviewer 2: I think it's good to have defaults for this two values.

Author: What would you recommend here as default?

Reviewer: I think we don`t need defaults here, because not every merchants is using custom order types
